src/staking/index.ts

```diff
@@ -444,4 +254,93 @@ export class Staking {
       );
     }
   }
+
+  /**
+   * Create a slash timelock unbonded transaction for staking.
+   * 
+   * @param {Transaction} stakingTx - The staking transaction to slash.
+   * without coordinates.
+   * 
+   * @returns {PsbtTransactionResult} - An object containing the unsigned psbt and fee
+   * 
+   * @throws {StakingError} - If the delegation is invalid or the transaction cannot be built
+   */
+  public createSlashTimelockUnbondedTransaction(
+    stakingTx: Transaction,
+  ) : PsbtTransactionResult {
+    if (!this.params.slashing) {
+      throw new StakingError(
+        StakingErrorCode.INVALID_PARAMS,
+        "Slashing parameters are missing",
+      );
+    }
+    const slashingAddress = deriveAddressFromPkScript(
+      this.params.slashing.slashingPkScript, this.network,
+    );
+    
+    // Build scripts
+    const scripts = this.buildScripts();
+
+    // create the slash timelock unbonded transaction
+    try {
+      const { psbt } = slashTimelockUnbondedTransaction(
+        scripts,
+        stakingTx,
+        slashingAddress,
+        this.params.slashing.slashingRate,
+        this.params.slashing.minSlashingTxFeeSat,
+        this.network,
+      );
+      return { psbt, fee: this.params.slashing.minSlashingTxFeeSat };
+    } catch (error) {
+      throw StakingError.fromUnknown(
+        error, StakingErrorCode.BUILD_TRANSACTION_FAILURE,
+        "Cannot build the slash timelock unbonded transaction",
+      );
+    }
+  }
+
+  /**
+   * Create a slash early unbonded transaction for staking.
+   * 
+   * @param {Transaction} unbondingTx - The unbonding transaction to slash.
+   * 
+   * @returns {PsbtTransactionResult} - An object containing the unsigned psbt and fee
+   * 
+   * @throws {StakingError} - If the delegation is invalid or the transaction cannot be built
+   */
+  public createSlashEarlyUnbondedTransaction(
+    unbondingTx: Transaction,
+  ): PsbtTransactionResult {
+    if (!this.params.slashing) {
+      throw new StakingError(
+        StakingErrorCode.INVALID_PARAMS,
+        "Slashing parameters are missing",
+      );
+    }
+    const slashingAddress = deriveAddressFromPkScript(
```

Comment:
If I read typescript correctly:
```
type PsbtOutputExtended = PsbtOutputExtendedAddress | PsbtOutputExtendedScript;
interface PsbtOutputExtendedAddress extends PsbtOutput {
    address: string;
    value: number;
}
interface PsbtOutputExtendedScript extends PsbtOutput {
    script: Buffer;
    value: number;
```
I think you could avoid this intermediate address derivation and pass slashing pk script as `Buffer` directly to `slashingTransaction` builder.